src/Sparrow.Server/Allocators.Pool.cs

```diff
@@ -0,0 +1,312 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Sparrow.Binary;
+using Sparrow.Global;
+
+namespace Sparrow.Server
+{
+    public interface IPoolAllocatorOptions : IAllocatorOptions, IComposableAllocator<Pointer>
+    {
+        int MaxBlockSize { get; }
+        int MaxPoolSizeInBytes { get; }
+    }
+
+    public static class PoolAllocator
+    {
+        public struct Default : IPoolAllocatorOptions, INativeOptions
+        {
+            public bool UseSecureMemory => false;
+            public bool ElectricFenceEnabled => false;
+            public bool Zeroed => false;
+
+            public int MaxBlockSize => 64 * Constants.Size.Megabyte;
+            public int MaxPoolSizeInBytes => 256 * Constants.Size.Megabyte;
+
+            public bool HasOwnership => true;
+
+            public IAllocatorComposer<Pointer> CreateAllocator()
+            {
+                var allocator = new Allocator<NativeAllocator<Default>>();
+                allocator.Initialize(default(Default));
+                return allocator;
+            }
+
+            /// <summary>
+            /// By default whenever we create an allocator we are going to dispose it too when the time comes.
+            /// </summary>
+            /// <param name="allocator">the allocator to dispose.</param>
+            public void ReleaseAllocator(IAllocatorComposer<Pointer> allocator, bool disposing)
+            {
+                allocator?.Dispose(disposing);
+            }
+        }
+
+        public struct Static : IPoolAllocatorOptions, INativeOptions
```

Comment:
This is identical do `Default`?

Reply:
No, the static one overrides the ReleaseAllocator disposing mechanism because it is design to deal with the `readonly static` lifecycle instead of a normal method stored allocator.